dlpar/dlpar_api/api.py

```diff
@@ -50,90 +47,47 @@ def __str__(self):
-        self.sendline("bind 'set enable-bracketed-paste off'")
-        self.expect('\r\n')
-        self.prompt()
-        self.sendline(command)
-        self.expect('\r\n')
-        self.prompt()
-        result = self.before.strip()
-        self.sendline('echo $?')
-        self.expect('\r\n')
-        self.prompt()
-        return_code = self.before.strip()
-        try:
-            int(return_code)
-        except ValueError:
-            self.__log('Got invalid return code. Trying again after 240s.')
-            time.sleep(240)
-            self.expect('\r\n')
-            self.prompt()
-            return_code = self.before.strip()
-            # If something goes wrong after 240 seconds, then it's better to
-            # throw an exception...
-            try:
-                int(return_code)
-            except ValueError:
-                e_msg = 'Got invalid return code again. Aborting.'
-                raise ValueError(e_msg)
-
-        self.__log("Command output: '%s'" % result)
-        self.__log("Return code: '%s'" % return_code)
-
-        if string and code:
-            return result, return_code
-        elif string:
-            return result
-        elif code:
-            return return_code
-
-    def __log(self, message):
-        """Just to use the log if we have one."""
-        if self.log:
-            self.log.debug(message)
-
-
-class SshMachine:
+class SshMachine():
     """The machine that we reach using the ssh protocol.
 
     This class take every important information about the machine
     and also the ssh connection.
     """
 
-    def __init__(self, machine_type, log=None):
+    def __init__(self, config_payload, machine_type, log=None):
         """Get every machine information."""
-        self.name = CONFIG_FILE.get(machine_type, 'name')
-        self.user = CONFIG_FILE.get(machine_type, 'user')
-        self.passwd = CONFIG_FILE.get(machine_type, 'passwd')
-        self.machine = CONFIG_FILE.get(machine_type, 'machine')
-        self.partition = CONFIG_FILE.get(machine_type, 'partition')
+        if machine_type == "linux_primary":
+            self.name = config_payload.get('src_name')
+            self.machine = config_payload.get('hmc_partition')
+            self.partition = config_payload.get('src_partition')
+        elif machine_type == "linux_secondary":
+            self.name = config_payload.get('target_lpar_name')
+            self.user = config_payload.get('target_user')
+            self.passwd = config_payload.get('target_passwd')
+            self.partition = config_payload.get('target_partition')
+            self.machine = config_payload.get('hmc_partition')
+        else:
+            self.name = config_payload.get('hmc_name')
+            self.user = config_payload.get('hmc_user')
+            self.passwd = config_payload.get('hmc_passwd')
+            self.partition = config_payload.get('hmc_partition')
+            self.machine = config_payload.get('hmc_partition')
```

Comment:
machine and partition are assigned same value. This seems incorrect. Can you check?

Reply:
Yes, Bothe variables are the same and are needed in multiple places.